### PR TITLE
fix: handle search params using `URLSearchParams`

### PR DESCRIPTION
**Pull request by yomybaby**
### TL;DR

Replaced string concatenation with `URLSearchParams` for constructing query parameters in several GET requests.

### What changed?

- Refactored various components such as `FolderExplorerOpener`, `ResourceGroupSelect`, `VFolderLazyView`, `VFolderSelect`, `VFolderTable`, and `backendai` hook to use `URLSearchParams` instead of manual string concatenation for query parameters.

### How to test?

1. Verify that the `FolderExplorerOpener` component correctly fetches folders using the updated URL search parameters.
2. Check that the `ResourceGroupSelect` component functions as expected with the new query parameter construction.
3. Ensure the `VFolderLazyView`, `VFolderSelect`, and `VFolderTable` components properly retrieve folder data with the updated URL structure.
4. Test the `backendai` hook's functionality, particularly the `/config/resource-slots/details` endpoint, to confirm the `URLSearchParams` integration.

### Why make this change?

To improve the readability, maintainability, and reliability of URL query parameter construction across the codebase by utilizing `URLSearchParams` instead of manual string concatenation.

---

The pull request refactors multiple components to use URLSearchParams for setting query parameters in backend API requests. This approach improves code consistency and readability. The affected components include FolderExplorerOpener, ResourceGroupSelect, VFolderLazyView, VFolderSelect, VFolderTable, and a hook in backendai for resource slot details.

---

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by yomybaby** [comment]
* **#2528** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2528?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yomybaby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/hooks/backendai.tsx [hunk 1]
```diff
@@ -73,16 +73,18 @@ export const useResourceSlotsDetails = (resourceGroupName?: string) => {
     queryFn: () => {
       // return baiClient.get_resource_slots();
       if (
-        _.isEmpty(resourceGroupName) ||
```
Comment (agatha197) [inline react/src/hooks/backendai.tsx:76]: Why did you change this line? lodash `isEmpty` can check empty string, null, undefined, etc.

**Review by agatha197 — CHANGES_REQUESTED** [review]

File: react/src/hooks/backendai.tsx (hunk 1 above)
Comment (ironAiken2) [reply react/src/hooks/backendai.tsx:76]: This was fixed by @yomybaby , so I don't know exactly why.

Comment (yomybaby) [reply react/src/hooks/backendai.tsx:76]: `resourceGroupName` is a string. !resourceGroupName is enough to determine if it is empty and better for TypeScript inference.

**Review by agatha197 — APPROVED** [review]
LGTM
